src/controllers/krate/search.rs

```diff
@@ -188,19 +227,538 @@ pub fn search(req: &mut Request) -> CargoResult<Response> {
             },
         )
         .collect();
+    crates
+}
 
-    #[derive(Serialize)]
-    struct R {
-        crates: Vec<EncodableCrate>,
-        meta: Meta,
+#[cfg(test)]
+mod test {
```

Comment:
Should this live with the rest of the integration tests, rather than re-implementing half of it here?

Reply:
Are you talking about the builder or the tests?

Reply:
Both